File: src/main/java/br/com/blu/bs2/sintetizador/contas/csv/service/strategy/IFactoryObterPercentualBonusService.java
package br.com.blu.bs2.sintetizador.contas.csv.service.strategy;


import br.com.blu.bs2.sintetizador.contas.csv.model.dominio.Conta;

import java.math.BigDecimal;

public interface IFactoryObterPercentualBonusService {

    boolean isAppliable(Conta conta);

    BigDecimal obterValorPercentualBonusAoDeposito(Conta conta);

    PercentualBonus getPercentualBonus(Conta conta);
}

File: src/main/kotlin/br/com/blu/bs2/sintetizador/contas/csv/service/strategy/IFactoryObterPercentualBonusService.kt
package br.com.blu.bs2.sintetizador.contas.csv.service.strategy

import br.com.blu.bs2.sintetizador.contas.csv.model.dominio.Conta
import java.math.BigDecimal

interface IFactoryObterPercentualBonusService {

    fun isAppliable(conta: Conta?): Boolean

    fun obterValorPercentualBonusAoDeposito(conta: Conta?): BigDecimal?

    fun getPercentualBonus(conta: Conta?): PercentualBonus?
}
